lightning/src/ln/channelmanager.rs

```diff
@@ -487,6 +487,22 @@ pub(super) struct PeerState<Signer: ChannelSigner> {
 	/// Messages to send to the peer - pushed to in the same lock that they are generated in (except
 	/// for broadcast messages, where ordering isn't as strict).
 	pub(super) pending_msg_events: Vec<MessageSendEvent>,
+	/// The peer is currently connected (i.e. we've seen a
+	/// [`ChannelMessageHandler::peer_connected`] and no corresponding
+	/// [`ChannelMessageHandler::peer_disconnected`].
+	is_connected: bool,
+}
+
+impl <Signer: ChannelSigner> PeerState<Signer> {
+	/// Indicates that a peer meets the criteria where we're ok to remove it from our storage.
+	/// If true is passed for `require_disconnected`, the function will return false if we haven't
+	/// disconnected from the node already, ie. `PeerState::is_connected` is set to `true`.
+	fn ok_to_remove(&self, require_disconnected: bool) -> bool {
```

Comment:
Are we ok with this function taking a passed `bool` to decide if `PeerState::is_connected` should be checked or not?

Another alternative would be to set `is_connected` to `false` in `ChannelManager::write` before serializing the `PeerState`, but that option becomes a bit problematic as we have some tests that triggers the `ChannelManager::write` function and after that attempts to call `ChannelManager::peer_disconnected` triggering a panic (specifically in `reload_tests::do_test_data_loss_protect`). So we'd then need to manually reset the value for `PeerState::is_connected ` to `true` after serializing but before calling `peer_disconnected` for such tests, which in my opinion is a worse option.

Let me know what you prefer!